pkgs/leak_tracker_flutter_testing/lib/src/leak_tracking_for_tests.dart

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:leak_tracker/leak_tracker.dart';
+import 'package:meta/meta.dart';
+
+void _emptyLeakHandler(Leaks leaks) {}
+
+/// Leak tracking settings and helper methods for tests.
+abstract class LeakTrackingForTests {
+  /// Current configuration for leak tracking.
+  static LeakTrackingForTestsSettings settings =
```

Comment:
Why is this a global state singleton? Shouldn't I be able to configure this for each test separately? And should this field be user-writable instead of requireing users to go through the defined methods below?

Reply:
Comment for the class may help.
I do not want users to create the settings from scratch, because I want defaults for package to affect all the test. So that if you set leak tracking to paused for the entire package, it would stop all the tests, except those that explicitly set paused=false, in spite of default.